pandas/tests/io/formats/test_format.py

```diff
@@ -3167,6 +3176,68 @@ def test_str(self):
         assert str(NaT) == "NaT"
 
 
+class TestPeriodIndexFormat:
+    def test_period(self):
+        """Basic test for period formatting with default format."""
+        p = pd.PeriodIndex([datetime(2003, 1, 1, 12), None], freq="H")
+        # default formatting
+        formatted = p.format()
+        assert formatted[0] == "2003-01-01 12:00"  # default: minutes not shown
+        assert formatted[1] == "NaT"
+        # format is equivalent to strftime(None)...
+        assert formatted[0] == p.strftime(None)[0]
+        assert p.strftime(None)[1] is np.nan  # ...except for NaTs
+
+        # Same test with nanoseconds freq
+        p = pd.period_range("2003-01-01 12:01:01.123456789", periods=2, freq="n")
+        formatted = p.format()
+        assert (formatted == p.strftime(None)).all()
+        assert formatted[0] == "2003-01-01 12:01:01.123456789"
+        assert formatted[1] == "2003-01-01 12:01:01.123456790"
+
+    def test_period_custom(self):
+        # GH46252
+
+        # Get locale-specific reference
+        am_local, pm_local = get_local_am_pm()
+
+        # 3 digits
+        p = pd.period_range("2003-01-01 12:01:01.123", periods=2, freq="l")
+        formatted = p.format(date_format="%y %I:%M:%S%p (ms=%l us=%u ns=%n)")
```

Comment:
Note: I expect this line to fail for now when run on locale `zh_CN`.
Once this is confirmed, I can either add the fix to #46319 in this branch, or remove the %p from this test so that it does not rely on locale.

Reply:
Strange that it did not fail. I will still not address the locale issue in this PR.